draco/core/io.py

```diff
@@ -352,6 +342,328 @@ def process(self):
+    """
+
+    def _parse_file(self, cfile):
+
+        from astropy.io import fits
+
+        with fits.open(cfile, mode="readonly") as cat:
+            pos = np.array(
+                [cat[1].data[col] for col in ["RA", "DEC"]], dtype=np.float64
+            ).T
+            pos = np.ascontiguousarray(pos).view(
+                [("ra", np.float64), ("dec", np.float64)]
+            )
+            z = cat[1].data["Z_DLA"][:]
+
+        # identify quasars with DLAs
+        has_dla = (z != -1).sum(axis=-1) > 0
+
+        # expand position array to cover redshift axis
+        pos = np.tile(pos[has_dla], (1, z.shape[1]))
+
+        # pick out cells with a DLA and collapse axes
+        is_dla = np.where(z[has_dla] != -1)
+        z = z[has_dla][is_dla]
+        pos = pos[is_dla]
+
+        # Apply any redshift selection to the objects
+        zsel = slice(None)
+        n = z.shape[0]
+        if self.z_range:
+            zsel = (z >= self.z_range[0]) & (z <= self.z_range[1])
+            n = np.sum(zsel)
+
+        # return an array with the expected shape
+        out = np.zeros((3, n), dtype=np.float64)
+        out[0, :] = pos["ra"][zsel]
+        out[1, :] = pos["dec"][zsel]
+        out[2, :] = z[zsel]
+
+        return out
+
+
+class LoadFITSCatalogDLA2(LoadFITSCatalog):
+    """Load DLA positions from a FITS source catalog structured like the one
+    provided in arXiv:2107.09612, which uses different column names than the
+    SDSS DR16 catalog.
+
+    Catalogs are given as one, or a list of `File Groups` (see
+    :mod:`draco.core.io`). Catalogs within the same group are combined together
+    before being passed on.
+
+    Attributes
+    ----------
+    catalogs : list or dict
+        A dictionary specifying a file group, or a list of them.
+    z_range : list, optional
+        Select only sources with a redshift within the given range.
+    freq_range : list, optional
+        Select only sources with a 21cm line freq within the given range. Overrides
+        `z_range`.
+    """
+
+    def _parse_file(self, cfile):
+
+        from astropy.io import fits
+
+        with fits.open(cfile, mode="readonly") as cat:
+            pos = np.array([cat[1].data[col] for col in ["RA", "DEC", "Z_CNN"]])
+
+        # Apply any redshift selection to the objects
+        if self.z_range:
+            zsel = (pos[2] >= self.z_range[0]) & (pos[2] <= self.z_range[1])
+            pos = pos[:, zsel]
+
+        return pos
+
+
+class LoadFITSDelta(task.SingleTask):
+    """Load eBOSS Lyman-alpha delta files in FITS format.
+
+    See https://data.sdss.org/datamodel/files/EBOSS_LYA/DELTA_LYAB/delta.html
+    for data format specification.
+
+    Attributes
+    ----------
+    delta_files : list or dict
+        A dictionary specifying a file group, or a list of them.
+    """
+
+    delta_files = config.Property(proptype=_list_of_filegroups)
+
+    def process(self):
+        """Load spectra from all files and combine into a single
```

Comment:
Initial docstring should be a single line. Longer comments can go in a paragraph below.